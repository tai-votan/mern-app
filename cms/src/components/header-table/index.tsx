import React, { useEffect } from 'react';
import { Button, Col, Dropdown, Form, Input, Menu, Row, Select } from 'antd';
import { FilterOutlined, SearchOutlined } from '@ant-design/icons';
import { useIntl, Link } from '@umijs/max';

interface HeaderTableProps {
  linkToNewPage: string;
  filterKey: Record<string, string>;
  filterValue: { label: string; value: string | boolean }[];
}

interface FormValues {
  keyword: string;
  filterKey: string;
  filterValue: string;
}

export const HeaderTable = (props: HeaderTableProps) => {
  const { formatMessage } = useIntl();
  const [form] = Form.useForm();
  const { linkToNewPage, filterKey, filterValue } = props;

  const onValuesChange = (_: FormValues, values: FormValues) => {
    console.log(`Func: onValuesChange - PARAMS: changedValue, values`, values);
  };

  useEffect(() => {
    form.setFieldValue('filterValue', filterValue[0].value);
  }, []);

  return (
    <Form
      form={form}
      onValuesChange={onValuesChange}
      initialValues={{
        filterKey: Object.keys(filterKey)[0],
      }}
    >
      <Row gutter={10}>
        <Col xs="auto">
          <Dropdown
            trigger={['click']}
            overlay={
              <Menu>
                <Menu.Item>
                  <Form.Item noStyle name="filterKey">
                    <Select
                      style={{ width: '100%' }}
                      placeholder="Please select a country"
                    >
                      {Object.keys(filterKey).map((key) => (
                        <Select.Option value={key} key={key}>
                          {formatMessage({ id: filterKey[key] })}
                        </Select.Option>
                      ))}
                    </Select>
                  </Form.Item>
                </Menu.Item>
                <Menu.Item>
                  <Form.Item noStyle name="filterValue">
                    <Select
                      style={{ width: '100%' }}
                      placeholder="Please select a country"
                    >
                      {filterValue.map((item) => (
                        <Select.Option
                          value={item.value}
                          key={`filter value ${item.label}`}
                        >
                          {formatMessage({ id: item.label })}
                        </Select.Option>
                      ))}
                    </Select>
                  </Form.Item>
                </Menu.Item>
              </Menu>
            }
            placement="bottomLeft"
          >
            <Button icon={<FilterOutlined />}>
              {formatMessage({ id: 'filter' })}
            </Button>
          </Dropdown>
        </Col>
        <Col flex="auto">
          <Form.Item name="keyword" noStyle>
            <Input
              prefix={<SearchOutlined />}
              placeholder={formatMessage({ id: 'search' })}
            />
          </Form.Item>
        </Col>
        <Col xs="auto">
          <Link to={linkToNewPage}>
            <Button type="primary">{formatMessage({ id: 'add page' })}</Button>
          </Link>
        </Col>
      </Row>
    </Form>
  );
};
