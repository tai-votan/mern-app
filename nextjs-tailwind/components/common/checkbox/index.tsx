import React, { useState } from "react";
import classNames from "classnames";

export interface CheckboxChangeEventTarget extends CheckboxProps {
  checked: boolean;
}

export interface CheckboxChangeEvent {
  target: CheckboxChangeEventTarget;
  stopPropagation: () => void;
  preventDefault: () => void;
  nativeEvent: MouseEvent;
}

export interface CheckboxProps {
  className?: string;
  defaultChecked?: boolean;
  checked?: boolean;
  style?: React.CSSProperties;
  disabled?: boolean;
  onChange?: (e: CheckboxChangeEvent) => void;
  onClick?: (e: React.MouseEvent<HTMLInputElement>) => void;
  value?: string;
  name?: string;
  children?: React.ReactNode;
  id?: string;
}

export const Checkbox: React.ForwardRefRenderFunction<HTMLInputElement, CheckboxProps> = (props) => {
  const { children, className, defaultChecked, disabled, onChange, ...restProps } = props;
  const [checked, setChecked] = useState<boolean>(props.checked || defaultChecked || false);

  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    if (disabled) {
      return;
    }
    if (!("checked" in props)) {
      setChecked(e.target.checked);
    }
    if (onChange) {
      onChange({
        target: {
          ...props,
          checked: e.target.checked,
        },
        stopPropagation() {
          e.stopPropagation();
        },
        preventDefault() {
          e.preventDefault();
        },
        nativeEvent: e.nativeEvent,
      });
    }
  };

  return (
    <label className="flex items-center">
      <input
        disabled={disabled}
        type="checkbox"
        checked={checked || defaultChecked}
        className={classNames("border outline-0 px-2 py-1.5 rounded cursor-pointer", className, {
          "cursor-not-allowed": disabled,
        })}
        onChange={(e) => handleChange(e)}
        {...restProps}
      />
      {children !== undefined && <span className="px-1.5">{children}</span>}
    </label>
  );
};
