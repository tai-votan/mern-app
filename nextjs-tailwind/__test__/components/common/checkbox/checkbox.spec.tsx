import { render } from "@testing-library/react";
import { Checkbox } from "@/components/common";

describe("Input", () => {
  it("renders input", () => {
    const { container } = render(<Checkbox />);
    expect(container).toMatchSnapshot();
  });

  it("renders input", () => {
    const { container } = render(<Checkbox>Label</Checkbox>);
    expect(container).toMatchSnapshot();
  });
});
