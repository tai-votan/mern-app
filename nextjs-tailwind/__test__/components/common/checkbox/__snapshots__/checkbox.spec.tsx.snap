// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders input 1`] = `
<div>
  <label
    class="flex items-center"
  >
    <input
      class="border outline-0 px-2 py-1.5 rounded cursor-pointer"
      type="checkbox"
    />
  </label>
</div>
`;

exports[`Input renders input 2`] = `
<div>
  <label
    class="flex items-center"
  >
    <input
      class="border outline-0 px-2 py-1.5 rounded cursor-pointer"
      type="checkbox"
    />
    <span
      class="px-1.5"
    >
      Label
    </span>
  </label>
</div>
`;
